app/src/main/java/org/cru/godtools/ui/tools/ToolsAdapter.kt

```diff
@@ -57,18 +58,13 @@ class ToolsAdapter(
 
     override fun getItemViewType(position: Int) = when (getItem(position)?.type) {
         Tool.Type.LESSON -> VIEW_TYPE_LESSON
-        else -> VIEW_TYPE_TOOL
+        else -> if (isAllTools) { VIEW_TYPE_ALL_TOOL } else { VIEW_TYPE_TOOL }
```

Comment:
instead of packing this into an if statement in the when else condition, flatten it out into the when statement itself